flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SlotSharingExecutionSlotAllocator.java

```diff
@@ -108,27 +122,30 @@
 
 		SharedSlotProfileRetriever sharedSlotProfileRetriever = sharedSlotProfileRetrieverFactory
 			.createFromBulk(new HashSet<>(executionVertexIds));
```

Comment:
Why is it enough to only put `executionVertexIds` in the set of `MergingSharedSlotProfileRetriever.allBulkPriorAllocationIds`? What if the failover includes multiple pipelined regions? Won't this mean that it can happen that the first pipelined regions takes slots from a preceding one?

Reply:
Indeed, this can result in suboptimal behaviour. I created an issue to investigate this: [FLINK-19142](https://issues.apache.org/jira/browse/FLINK-19142).